### ISSUE TITLE
Incorrect `Expecting directive-name but found "↩"`

### ISSUE THREAD

**Issue by wdanilo** [validator/validator#917]
Hi! I believe the error I get for the following HTML input is incorrect:

```html
<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="Content-Security-Policy" content="
            default-src 'self';
            script-src  'self' 'unsafe-eval';
            style-src   'self' 'unsafe-inline';
            connect-src 'self' data:;
        "/>
        <meta name="viewport" content="
            width=device-width,
            initial-scale = 1.0,
            maximum-scale = 1.0,
            user-scalable = no
        ">
        <title>Enso IDE</title>
        <style>
          html, body { width:100vw; height:100vh; }
          body       { margin:0; }
        </style>
        <script type="module" src="/assets/index.js"></script>
    </head>
    <body style="background-color:#e8e7e6">
        <noscript>This page requires JavaScript to run. Please enable it in your browser.</noscript>
    </body>
</html>
```

Error:

```
Error: Bad value ↩            default-src 'self';↩            script-src  'self' 'unsafe-eval';↩            style-src   'self' 'unsafe-inline';↩            connect-src 'self' data:;↩         for attribute content on element meta: Expecting directive-name but found "↩". Expecting directive-name but found "↩". Expecting directive-name but found "↩". Expecting directive-name but found "↩". Expecting directive-name but found "↩".
```

**Comment by nschonni** [comment]
I don't think newlines are valid in any HTML attribute value

**Comment by sideshowbarker** [comment]
Newlines are OK in most attribute values, actually. Whether they’re allowed for a particular attribute depends on what microsyntax requirements are defined for that attribute, and what spec defines those requirements.

For the case of the `content` value for an `http-equiv="Content-Security-Policy"` meta element, the requirements are set by the CSP spec — specifically, here:

https://w3c.github.io/webappsec-csp/#grammardef-serialized-policy

> **serialized-policy** =
>    serialized-directive *( optional-ascii-whitespace ";" [ optional-ascii-whitespace serialized-directive ] )

…where **optional-ascii-whitespace** is defined as:

> *( %x09 / %x0A / %x0C / %x0D / %x20 )

Given that, as far as I can see, no error should be reported for the policy shown in the question.

So I think this might be a bug in the Salvation CSP library that we use for CSP policy checking. I’ll raise an issue in their tracker and see what they say.

**Comment by sideshowbarker** [comment]
Raised https://github.com/shapesecurity/salvation/issues/234

**Comment by sideshowbarker** [comment]
OK, it seems this is in fact a bug a Salvation, per the comment at https://github.com/shapesecurity/salvation/issues/234#issuecomment-593198520 from one of the Salvation developers.

So, nice catch — and thanks for having taken time to report this. 

For the real fix, we’ll need to wait until the Salvation code gets changed upstream, and they do a new release. But for now, I’ve added a workaround to the HTML-checker code, and pushed the changed to https://validator.w3.org/nu/
